### DISCUSSION TITLE
Libraries and runtime updates in .NET 9 Preview 7

### DISCUSSION THREAD

**Discussion by richlander** [dotnet/runtime#106350] [Announcements]
Lots new this release to try out.

## **📚Libraries**

- [Removal of `BinaryFormatter` is complete](https://github.com/dotnet/core/blob/main/release-notes/9.0/preview/preview7/libraries.md#removal-of-binaryformatter-is-complete)
- [Enumerate over `ReadOnlySpan<char>.Split()` segments](https://github.com/dotnet/core/blob/main/release-notes/9.0/preview/preview7/libraries.md#enumerate-over-readonlyspancharsplit-segments)
- [`Debug.Assert` now reports assert condition, by default.](https://github.com/dotnet/core/blob/main/release-notes/9.0/preview/preview7/libraries.md#debugassert-now-reports-assert-condition-by-default)
- [Compression APIs now use `zlib-ng`](https://github.com/dotnet/core/blob/main/release-notes/9.0/preview/preview7/libraries.md#compression-apis-now-use-zlib-ng)
- [`Guid.CreateVersion7` enables creating GUIDs with a natural sort order](https://github.com/dotnet/core/blob/main/release-notes/9.0/preview/preview7/libraries.md#guidcreateversion7-enables-creating-guids-with-a-natural-sort-order)
- [`Interlocked.CompareExchange` for more types](https://github.com/dotnet/core/blob/main/release-notes/9.0/preview/preview7/libraries.md#interlockedcompareexchange-for-more-types)
- [AES-GCM and ChaChaPoly1305 algorithms enabled for iOS/tvOS/MacCatalyst](https://github.com/dotnet/core/blob/main/release-notes/9.0/preview/preview7/libraries.md#aes-gcm-and-chachapoly1305-algorithms-enabled-for-iostvosmaccatalyst)
- [Changes to X.509 Certificate Loading](https://github.com/dotnet/core/blob/main/release-notes/9.0/preview/preview7/libraries.md#changes-to-x509-certificate-loading)
- [Support for XPS documents from XPS virtual printer](https://github.com/dotnet/core/blob/main/release-notes/9.0/preview/preview7/libraries.md#support-for-xps-documents-from-xps-virtual-printer)
- [Marking `Tensor<T>` as `Experimental`](https://github.com/dotnet/core/blob/main/release-notes/9.0/preview/preview7/libraries.md#marking-tensort-as-experimental)
- Full [release notes](https://github.com/dotnet/core/blob/main/release-notes/9.0/preview/preview7/libraries.md)

## **⏱️Runtime**

- [ARM64 SVE Support](https://github.com/dotnet/core/blob/main/release-notes/9.0/preview/preview7/runtime.md#arm64-sve-support)
- [Post-Indexed Addressing on ARM64](https://github.com/dotnet/core/blob/main/release-notes/9.0/preview/preview7/runtime.md#post-indexed-addressing-on-arm64)
- [Strength Reduction in Loops](https://github.com/dotnet/core/blob/main/release-notes/9.0/preview/preview7/runtime.md#strength-reduction-in-loops)
- [Object Stack Allocation for Boxes](https://github.com/dotnet/core/blob/main/release-notes/9.0/preview/preview7/runtime.md#object-stack-allocation-for-boxes)
- [GC Dynamic Adaptation To Application Sizes](https://github.com/dotnet/core/blob/main/release-notes/9.0/preview/preview7/runtime.md#gc-dynamic-adaptation-to-application-sizes)
- Full [release notes](https://github.com/dotnet/core/blob/main/release-notes/9.0/preview/preview7/runtime.md)

Release notes:

- [Libraries release notes](https://github.com/dotnet/core/blob/main/release-notes/9.0/preview/preview7/libraries.md)
- [Runtime release notes](https://github.com/dotnet/core/blob/main/release-notes/9.0/preview/preview7/runtime.md)

.NET 9 Preview 7:

- [Discussion](https://aka.ms/dotnet/9/preview7)
- [Release notes](https://github.com/dotnet/core/blob/main/release-notes/9.0/preview/preview7/README.md)
- [What's new in .NET 9](https://learn.microsoft.com/dotnet/core/whats-new/dotnet-9/overview)

**Comment by ReneeGA2020** [comment]
wasm import export sample seems to be broken in this preview:
https://github.com/dotnet/runtime/tree/main/src/mono/sample/wasi/native

I got this error when try to ran the sample.

> Error: cannot execute a component without `--wasm component-model`

Seems like the wasmtime recognize the output wasm file to be a component instead of executable wasm file.

**Reply by richlander** [reply]
@lewing

**Reply by pavelsavara** [reply]
@ReneeGA2020 note that WASI target is quite immature experiment, not a supported target. It's not part of the Net9 release and not really part of the Net9 previews. 

We are making some more progress on main branch of the repository and we don't plan to backport fixes to Net9. If you don't want to build your own runtime from sources, you are probably better off by downloading one of the [nightly builds](https://aka.ms/dotnet/10.0.1xx/daily/dotnet-sdk-win-x64.zip)

Regarding this particular issue is probably caused by old version of wasmtime installed on your machine, but I'm not sure. We switched to producing WASI preview2, component model binaries some time ago. 

Anyway, this preview7 announcement is not right place to troubleshoot it. 
Please feel free to open separate issue and ping me on it, thank you.

**Reply by ReneeGA2020** [reply]
Oh, thank you, this issue was actually addressed in another tick 2 weeks ago. I forgot to update here.
https://github.com/dotnet/runtime/issues/106523

Thank for the .net team's work.
